contrib/side-cars/fee-estimate.sh

```diff
@@ -0,0 +1,216 @@
+#!/bin/bash
+
+####################################
+# Usage
+# 
+# $ # one-shot fee-rate calculation
+# $ ./fee-estimate.sh
+# 161
+# 
+# $ # Check fees every 5 seconds and update `satoshis_per_byte` in `/path/to/miner.toml`
+# $ ./fee-estimate.sh watch /path/to/miner.toml 5
+# 
+# $ # Run unit tests and report result (0 means success)
+# $ ./fee-estimate.sh test; echo $?
+# 0
+####################################
+
+set -uoe pipefail
+
+function exit_error() {
+   echo >&2 "$@"
+   exit 1
+}
+
+####################################
+# Dependencies
+####################################
+# Dependencies
+for cmd in curl jq bc sed date grep; do
+   command -v "$cmd" >/dev/null 2>&1 || exit_error "Command not found: '$cmd'"
+done
+
+
+####################################
+# Functions
+####################################
+
+# Convert a fee/kb to fee/vbyte.
+# If there's a fractional part of the fee/kb (i.e. if it's not divisible by 1000),
+# then round up.
+# Arguments:
+#   $1 -- the fee per kb
+# Stdout: the satoshis per vbyte, as an integer
+# Stderr: none
+# Return:
+#   0 on success
+#   nonzero on error
+function fee_per_kb_to_fee_per_vbyte() {
+   local fee_per_kb="$1"
+   local fee_per_vbyte_float=
+   local fee_per_vbyte_ipart=
+   local fee_per_vbyte_fpart=
+   local fee_per_vbyte=
+
+   # must be an integer
+   if ! [[ "$fee_per_kb" =~ ^[0-9]+$ ]]; then
+      exit_error "Did not receive a fee/kb from $fee_endpoint, but got '$fee_per_kb'"
+   fi
+
+   # NOTE: round up -- get the fractional part, and if it's anything other than 000, then add 1
+   fee_per_vbyte_float="$(echo "scale=3; $fee_per_kb / 1000" | bc)"
+   fee_per_vbyte_ipart="$(echo "$fee_per_vbyte_float" | sed -r 's/^([0-9]*)\..+$/\1/g')"
+   fee_per_vbyte_fpart="$(echo "$fee_per_vbyte_float" | sed -r -e 's/.+\.([0-9]+)$/\1/g' -e 's/0//g')"
+   fee_per_vbyte="$fee_per_vbyte_ipart"
+   if [ -n "$fee_per_vbyte_fpart" ]; then
+      fee_per_vbyte="$((fee_per_vbyte + 1))"
+   fi
+
+   echo "$fee_per_vbyte"
+   return 0
+}
+
+# Determine satoshis per vbyte
+# Arguments: none
+# Stdout: the satoshis per vbyte, as an integer
+# Stderr: none
+# Return:
+#   0 on success
+#   nonzero on error
+function get_sats_per_vbyte() {
+   local fee_endpoint="https://api.blockcypher.com/v1/btc/main"
+   local fee_per_kb=
+
+   fee_per_kb="$(curl -sL "$fee_endpoint" | jq -r '.high_fee_per_kb')"
```

Comment:
Recommend a little more verbose status code check:
```suggestion
   local response="$(curl -sL -w "\n%{http_code}" $fee_endpoint)";
   local http_status_code="$(echo "$response" | tail -n 1)";
   case $http_status_code in 
      200)
         ;;
      429)
         exit_error "Rate-Limited"
         ;;
      404)
         exit_error "404 error retrieving ${fee_endpoint}"
         ;;
      **)
         exit_error "${http_status_code} Error retrieving ${fee_endpoint}"
         ;;
   esac
   fee_per_kb="$(echo "$response" | sed '$d' | jq .high_fee_per_kb)";
```

ex:
```
 ./sidecar.sh
404 error retrieving https://api.blockcypher.com/v1/btc/mainasdf
```

Reply:
Done